adapters/verizonmedia/verizonmedia.go

```diff
@@ -160,13 +165,35 @@ func getImpInfo(impId string, imps []openrtb.Imp) (bool, openrtb_ext.BidType) {
 	return exists, mediaType
 }
 
-func changeRequestForBidService(request *openrtb.BidRequest, extension *openrtb_ext.ExtImpVerizonMedia) {
+func changeRequestForBidService(request *openrtb.BidRequest, extension *openrtb_ext.ExtImpVerizonMedia) error {
 	if request.Imp[0].TagID == "" {
 		request.Imp[0].TagID = extension.Pos
 	}
 	if request.Site.ID == "" {
 		request.Site.ID = extension.Dcn
 	}
+
+	if request.Imp[0].Banner == nil {
+		return nil
+	}
+
+	banner := *request.Imp[0].Banner
+	request.Imp[0].Banner = &banner
+
+	if banner.W != nil && banner.H != nil {
+		return nil
+	}
+
+	if len(banner.Format) == 0 {
+		return errors.New(fmt.Sprintf("No sizes provided for Banner %v", banner.Format))
+	}
+
+	banner.W = new(uint64)
+	*banner.W = banner.Format[0].W
+	banner.H = new(uint64)
+	*banner.H = banner.Format[0].H
+
+	return nil
```

Comment:
I know some of the scenarios listed below may be unlikely for an incoming bid request, but I guess it's better to be safe than sorry. Current logic as written right now, unless I'm wrong in the table below, should yield the result listed in the rightmost column.
```
   | banner.W | banner.H | len(banner.Format)>0 | banner.Format[0].W | banner.Format[0].H  ||         RESULT
-----------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 |   nil    |   nil    |      false           |       ---          |       ---           || Error "No sizes provided for Banner %v", banner.Format
 2 |   nil    |   nil    |      false           |       ---          |       ---           || Error "No sizes provided for Banner %v", banner.Format
 3 |   nil    |   nil    |      false           |       ---          |       ---           || Error "No sizes provided for Banner %v", banner.Format
 4 |   nil    |   nil    |      false           |       ---          |       ---           || Error "No sizes provided for Banner %v", banner.Format
 5 |   nil    |   nil    |       true           | 0 or invalid value | 0 or invalid value  || banner.W = 0, banner.H = 0    // Don't we want to throw an error?
 6 |   nil    |   nil    |       true           | 0 or invalid value |        90           || banner.W = 0, banner.H = 90   // Don't we want to throw an error?
 7 |   nil    |   nil    |       true           |      720           | 0 or invalid value  || banner.W = 720, banner.H = 0  // Don't we want to throw an error?
 8 |   nil    |   nil    |       true           |      720           |        90           || banner.W = 720, banner.H = 90
 9 |   nil    |    90    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
10 |   nil    |    90    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
11 |   nil    |    90    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
12 |   nil    |    90    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format 
13 |   nil    |    90    |       true           | 0 or invalid value |  0 or invalid value || banner.W = 0, banner.H = 0    // Don't we want to throw an error?
14 |   nil    |    90    |       true           | 0 or invalid value |        90           || banner.W = 0, banner.H = 90   // Don't we want to throw an error?
15 |   nil    |    90    |       true           |      720           | 0 or invalid value  || banner.W = 720, banner.H = 0  // Do we want to use the "90" value so banner.W = 720, banner.H = 90 ?
16 |   nil    |    90    |       true           |      720           |        90           || banner.W = 720, banner.H = 90
17 |   720    |   nil    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
18 |   720    |   nil    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
19 |   720    |   nil    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
20 |   720    |   nil    |      false           |      ---           |       ---           || Error "No sizes provided for Banner %v", banner.Format
21 |   720    |   nil    |       true           | 0 or invalid value | 0 or invalid value  || banner.W = 0, banner.H = 0    // Don't we want to throw an error?
22 |   720    |   nil    |       true           | 0 or invalid value |        90           || banner.W = 0, banner.H = 90   // Don't we want to use the "720" value so banner.W = 720, banner.H = 90 ?
23 |   720    |   nil    |       true           |      720           | 0 or invalid value  || banner.W = 720, banner.H = 0  // Don't we want to throw an error?
24 |   720    |   nil    |       true           |      720           |        90           || banner.W = 720, banner.H = 90
25 |   720    |    90    |      false           |      ---           |       ---           || banner.W = 720, banner.H = 90
26 |   720    |    90    |      false           |      ---           |       ---           || banner.W = 720, banner.H = 90
27 |   720    |    90    |      false           |      ---           |       ---           || banner.W = 720, banner.H = 90
28 |   720    |    90    |      false           |      ---           |       ---           || banner.W = 720, banner.H = 90
29 |   720    |    90    |       true           | 0 or invalid value | 0 or invalid value  || banner.W = 720, banner.H = 90         
30 |   720    |    90    |       true           | 0 or invalid value |        90           || banner.W = 720, banner.H = 90         
31 |   720    |    90    |       true           |      720           | 0 or invalid value  || banner.W = 720, banner.H = 90         
32 |   720    |    90    |       true           |      720           |                     || banner.W = 720, banner.H = 90         
```
Don't you think scenarios were one or both dimensions are "0" (or any other non-useful value for that matter) should also return an error? Just double checking here.

Reply:
I added an extra check when the banner.H or banner.W are zero.
What do you mean by "invalid value" in the banner.format and how can I simulate that?. What I see is that the banner.format array is validated, you can only put there a positive value non-zero value. So in the adapter the format array should end-up having positive "valid values".
Also I would not combine parts of format array entries with the values of the banner.H or banner.W potentially create a different size, like the example 22 you gave. 
If the request builder does not set the banner.H and banner.W both values with right values but they do that in the banner.format I think is correct to use that format entry.

Reply:
Got it